src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -1565,6 +1579,60 @@ private static boolean withinRange(RepairSegment segment, Optional<RingRange> ra
     return !range.isPresent() || segmentIsWithinRange(segment, range.get());
   }
 
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions() {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public Collection<DiagEventSubscription> getEventSubscriptions(String clusterName) {
+    return session.execute(getDiagnosticEventsPrepStmt.bind()).all().stream()
+        .map((row) -> createDiagEventSubscription(row))
+        .filter((subscription) -> clusterName.equals(subscription.getCluster()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public DiagEventSubscription getEventSubscription(UUID id) {
+    return createDiagEventSubscription(session.execute(getDiagnosticEventPrepStmt.bind(id)).one());
+  }
+
+  private static DiagEventSubscription createDiagEventSubscription(Row row) {
+    return new DiagEventSubscription(
+        Optional.of(row.getUUID("id")),
+        row.getString("cluster"),
+        Optional.of(row.getString("description")),
+        row.getList("include_nodes", String.class),
+        row.getList("events", String.class),
+        row.getBool("export_sse"),
+        row.getString("export_file_logger"),
+        row.getString("export_http_endpoint"));
+  }
+
+  @Override
+  public DiagEventSubscription addEventSubscription(DiagEventSubscription subscription) {
+    session.execute(saveDiagnosticEventPrepStmt.bind(
+        subscription.getId(),
```

Comment:
`subscription.getId()` is returning an `Optional<UUID>` while the query is expecting a `UUID`.
I've tried to add `.get()` but the id was null when creating new subscriptions. It ended up working after replacing this line with : 

```
        subscription.getId().orElse(UUID.randomUUID()),
```

Reply:
Thanks for spotting this.  The transformation was there already in `MemoryStorage`. I pushed it up to `DiagEventSubscriptionService`, and in both storages put int a precondition check.